### PR TITLE
Hotfix/2017 06 20 iss1296 article dedupe

### PR DESCRIPTION
**Pull request by Steven-Eardley**
This is the first half of the article de-duplication work, covering improvements to the tests and a refactor of fetching duplicates.

This doesn't extend into the script required for #1296. It'll be easier to check that one based on this one.

### CODE REVIEW

File: portality/scripts/remove_duplicate_articles.py [hunk 1]
```diff
@@ -46,15 +61,19 @@
         if duplicates:
```
Comment (richard-jones) [inline portality/scripts/remove_duplicate_articles.py]: This line: XWalk().get_duplicate(a, all_duplicates=True)

all_duplicates is no longer part of the signature for that method, you want XWalk().get_duplicates() instead

File: portality/scripts/remove_duplicate_articles.py [hunk 2]
```diff
@@ -30,13 +39,19 @@
...
+        print 'Starting {0} {snapshot} in 5 seconds, Ctrl+C to exit.'.format(start.isoformat(), snapshot=snapshot_report)
     time.sleep(5)
 
+    count = 0
     originals = []  # make sure we only delete FORWARD. So, if articles
-    # A, B and C are duplicates, we stumble upon A first, and delete B and C.
+    # A, B and C are duplicates, we stumble upon A first, and delete B and C.cat
     # We must then ensure we do not delete A if we chance upon B or C
     # (because they are still in Python memory even though they are not in ES).
     article_iterator = models.Article.iterall(page_size=5000)
```
Comment (richard-jones) [inline portality/scripts/remove_duplicate_articles.py]: I seem to remember something about the order that things get considered.  I presume we want A to be the actual one to keep (e.g. the most recent, or the oldest, whatever the rule is).  iterall doesn't inherently order anything, I don't think, or if it does it is by ID.  So your ordering here is not guaranteed.

Comment (Steven-Eardley) [reply portality/scripts/remove_duplicate_articles.py]: This script has been completely rewritten in the part 2 pull request #1387 - it doesn't use an iterator any more, but a scroll. I should've really made this PR just to cover the changes in ```article.py```

**Comment by Steven-Eardley** [comment]
Since the run script is almost unrecognisable in the next PR, I'm going to count your review of this one as OK!
